File: scripts/test_company_intelligence_v2.py
#!/usr/bin/env python3
"""Regression tests for Company Intelligence V2 migration and public materialization."""

from __future__ import annotations

from copy import deepcopy

from materialize_company_intelligence_v2 import materialize_companies, public_v2_payload
from migrate_company_intelligence_v2 import DEFAULT_COMPANIES, build_v2_dataset, protected_hash, read_json


def main() -> int:
    legacy = read_json(DEFAULT_COMPANIES)
    payload = build_v2_dataset(DEFAULT_COMPANIES)
    assert len(payload["companies"]) == 17
    assert len({item["company_id"] for item in payload["companies"]}) == 17
    assert len({item["fact_id"] for item in payload["facts"]}) == len(payload["facts"])
    assert len({item["event_id"] for item in payload["events"]}) == len(payload["events"])
    assert len({item["source_id"] for item in payload["evidence"]}) == len(payload["evidence"])

    corrected = next(item for item in payload["events"] if item["event_id"] == "event-yuchang-enc-samsung-ai-modular-home")
    assert corrected["event_type"] == "partnership"
    assert corrected["event_status"] == "not_signed"
    assert corrected["project_credit"] is False
    assert corrected["verification_status"] == "not_verified"
    assert len(corrected["source_ids"]) == 35
    assert not any(item["project_credit"] for item in payload["events"] if item["event_type"] in {"partnership", "mou"})

    credited = [item for item in payload["events"] if item["project_credit"]]
    assert len(credited) == 1
    assert all(item["event_type"] == "project" for item in credited)
    assert all(item["project_role"] not in {None, "", "unknown", "role_unknown"} for item in credited)

    public = public_v2_payload(payload)
    assert len(public["companies"]) == 17
    assert all(item.get("visibility") == "public" for key in ["companies", "facts", "events", "evidence", "corrections"] for item in public[key])
    assert not any(item.get("visibility") == "internal" for key in ["companies", "facts", "events", "evidence", "corrections"] for item in public[key])
    yuchang_summary = next(item for item in public["materialized_summaries"] if item["company_id"] == "yuchang-enc")
    assert yuchang_summary["event_counts"]["verified_projects"] == 0
    assert yuchang_summary["event_counts"]["partnerships_mou"] == 1
    assert yuchang_summary["article_evidence_count"] == 35
    assert yuchang_summary["project_status"] if "project_status" in yuchang_summary else True

    before = {company["company_id"]: protected_hash(company) for company in legacy["companies"]}
    materialized = materialize_companies(deepcopy(legacy), public, payload)
    after = {company["company_id"]: protected_hash(company) for company in materialized["companies"]}
    assert before == after
    assert all(company.get("intelligence_v2", {}).get("summary_ko") for company in materialized["companies"])
    assert all(company.get("intelligence_v2", {}).get("domain_statuses") for company in materialized["companies"])
    assert sum(company.get("intelligence_v2", {}).get("event_counts", {}).get("verified_projects", 0) for company in materialized["companies"]) == 1

    print("COMPANY INTELLIGENCE V2 TESTS PASSED")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: frontend/scripts/test-company-intelligence-v2.mjs
import assert from "node:assert/strict";
import { readFileSync } from "node:fs";
import {
  getCompanyDataStatusLabel,
  getCompanyDomainStatuses,
  getCompanyEvents,
  getCompanyItems,
  getCompanyProjectSummary,
  getCompanySourceGroups,
  getKoreanCompanySummary,
} from "../src/companyInsights.js";

const companiesPayload = JSON.parse(readFileSync(new URL("../public/data/companies/companies.json", import.meta.url), "utf8"));
const v2 = JSON.parse(readFileSync(new URL("../public/data/companies/company_intelligence_v2.json", import.meta.url), "utf8"));
const companies = getCompanyItems(companiesPayload);
const byId = (id) => companies.find((company) => company.company_id === id);
const verifiedIds = new Set([
  "gs-ec", "hyundai-engineering", "samsung-ct-construction", "dl-enc",
  "yuchang-enc", "kumkang-kind", "nrb", "planm", "geogwang-enterprise", "sungji-steel",
]);

assert.equal(v2.schema_version, "2.0.0");
assert.equal(companies.length, 11);
assert.equal(companies.filter((company) => company.company_id === "daeseung-engineering").length, 1);
assert.equal(v2.companies.length, 10);
assert.equal(v2.materialized_summaries.length, 10);
assert.equal(companies.every((company) => company.intelligence_v2?.domain_statuses), true);
assert.equal(companies.every((company) => /[가-힣]/.test(getKoreanCompanySummary(company))), true);
assert.equal(companies.every((company) => !getCompanyDataStatusLabel(company).includes("_")), true);
assert.equal(v2.facts.every((record) => record.visibility === "public"), true);
assert.equal(v2.events.every((record) => record.visibility === "public"), true);
assert.equal(v2.evidence.every((record) => record.visibility === "public"), true);
assert.equal(v2.corrections.every((record) => record.visibility === "public"), true);

for (const id of verifiedIds) {
  assert.equal(byId(id).intelligence_v2.overall_data_status, "core_verified");
  assert.equal(getCompanyDomainStatuses(byId(id)).identity_status, "cross_verified");
  assert.ok(getCompanySourceGroups(byId(id)).some((group) => group.sources.some((source) => source.source_id === `manual-verified-${id}-20260716`)));
}

const yuchang = byId("yuchang-enc");
const legacySamsung = getCompanyEvents(yuchang).find((event) => event.event_id === "event-yuchang-enc-samsung-ai-modular-home");
assert.equal(legacySamsung.event_type, "partnership");
assert.equal(legacySamsung.event_status, "not_signed");
assert.equal(legacySamsung.project_credit, false);
assert.equal(getCompanyProjectSummary(yuchang).verified, 9);
assert.equal(getCompanyProjectSummary(yuchang).partnerships, 1);

const planmEvents = getCompanyEvents(byId("planm"));
assert.equal(planmEvents.find((event) => event.event_id === "event-planm-jindo-baseball-precon").project_credit, false);
assert.equal(planmEvents.find((event) => event.event_id === "event-planm-indiana-l7-precon").project_credit, false);
assert.equal(getCompanyProjectSummary(byId("planm")).verified, 10);
assert.equal(getCompanyProjectSummary(byId("planm")).candidates, 7);

const credited = v2.events.filter((event) => event.project_credit);
assert.ok(credited.length >= 40);
assert.equal(credited.every((event) => event.event_type === "project"), true);
assert.equal(v2.evidence.filter((item) => item.source_type === "manual_verified_research").length, 10);
assert.equal(v2.evidence.filter((item) => item.source_type === "media_article").length, 35);

const factIds = v2.facts.map((item) => item.fact_id);
const eventIds = v2.events.map((item) => item.event_id);
const evidenceIds = v2.evidence.map((item) => item.source_id);
assert.equal(factIds.length, new Set(factIds).size);
assert.equal(eventIds.length, new Set(eventIds).size);
assert.equal(evidenceIds.length, new Set(evidenceIds).size);

console.log("COMPANY INTELLIGENCE V2 FRONTEND TESTS PASSED");
